packages/trpc/server/routers/viewer/slots/util.ts

```diff
@@ -424,9 +426,44 @@ export async function getAvailableSlots({ input, ctx }: GetScheduleOptions): Pro
       },
       select: {
         userId: true,
+        attendees: {
+          select: {
+            email: true,
+          },
+        },
       },
     });
-    hosts = hosts.filter((host) => host.user.id === originalRescheduledBooking?.userId || 0);
+    if (
+      eventType.rescheduleWithSameRoundRobinHost &&
+      eventType.schedulingType === SchedulingType.ROUND_ROBIN
+    ) {
+      hosts = hosts.filter((host) => host.user.id === originalRescheduledBooking?.userId || 0);
+    }
+
+    const session = ctx?.req ? await getServerSession({ req: ctx?.req }) : null;
+    const userId = session?.user?.id;
+    const isUserEventOwner = isUserReschedulingOwner(userId, originalRescheduledBooking?.userId || 0)
+    const membership = eventType?.team?.members.find((membership) => membership.userId === userId);
+    const isUserTeamAdminOrOwner = membership?.role === MembershipRole.OWNER || membership?.role === MembershipRole.ADMIN;
```

Comment:
This covers the scenario for team owners but would fail if current user is org owners but they are not in the same team as event owner.